### PR TITLE
Fix for CI in 1.5.x branch

### PR DESCRIPTION
**Pull request by dmgav**
Fixed CI for v1.5.x branch:
- removed `test_reg_reader` test (uses deprecated feature of Databroker);
- fixed `test_table` test.

### CODE REVIEW

**Comment by danielballan** [comment]
Cherry-picked @tacaswell's commit from `master` that fixed the 3.7 build.

**Comment by danielballan** [comment]
That helped, but all the 0MQ tests are failing.

**Comment by dmgav** [comment]
The branch used to be tested only for Python 3.6. All tests are passing locally with Python 3.7.

**Comment by danielballan** [comment]
I can reproduce the Python 3.7 `test_zmq.py` failures locally.

We pinned pyzmq to 16.0.4 in 552a1b0e8 when 17.0.0 broke our CI. (We never discovered what the exact problems was. Then we unpinned it in cfc9c2a59 "to fix py37" as it says in the commit message. I have pushed a new commit to this branch to unpin that.

**Comment by danielballan** [comment]
OK, with that fix 3.7 passes. 3.8 has a slew of failures. Any objection to removing 3.8 from the 1.5.x test matrix. (The `master` branch tests on 3.8, so we'll support it for 1.6.0+.)

**Comment by dmgav** [comment]
I removed tests with Python 3.8. Hopefully everyone will quickly migrate to v1.6.0.

File: .travis.yml [hunk 1]
```diff
@@ -12,12 +12,10 @@ cache:
 matrix:
   fast_finish: true
   include:
-    - python: 3.6
-      env: BUILD_DOCS=1
```
Comment (mrakitin) [inline .travis.yml]: I am concerned by the removal of this line to force it to build docs. Is it intended?

Comment (mrakitin) [reply .travis.yml]: The commit message for https://github.com/bluesky/bluesky/pull/1309/commits/f526e3be4910acfc497be298456380701db92296 explains why it was done. It's OK to not build the docs for the maintenance branch.

**Comment by danielballan** [comment]
OK. Supporting the current latest stable Python would normally be standard practice, but given the circumstances I think we should carry on and address 3.8 only if there is a need.

**Comment by dmgav** [comment]
I removed the flag that enables building docs because it looks like it could overwrite docs on `blueskyproject.io`. If I am wrong and this is not the case, then I can enable document building.

**Comment by danielballan** [comment]
Seems correct to me. Until recently we haven't had much need for back-port releases so, speaking for myself, I'm still sorting out how to do it properly.

**Pull request merged**